### PR TITLE
fix(cheminfo-publish): show error log when npm publish fails

### PR DESCRIPTION
**Pull request by stropitek**
I published generator-cheminfo with those changes and it worked fine

Was able to see npm publish error, fix and then publish

### CODE REVIEW

**Comment by stropitek** [comment]
`execa()` does not work with commands without specifying the full path

You have to use `execa.shell()`

**Comment by stropitek** [comment]
Sorry for the bad diff... Prettier...

**Comment by targos** [comment]
> execa() does not work with commands without specifying the full path

That's not true. See how I used it here for example:

https://github.com/nodejs/node-core-utils/blob/74a3a8991de444576a6a04f5f863971c6f9e611a/lib/update-v8/updateV8Clone.js#L23

**Comment by stropitek** [comment]
I assumed because I had an ENOENT error when trying it with npm

**Comment by targos** [comment]
```
> execa('npm', ['--version']).then(console.log)
Promise {
  <pending>,
  domain:
   Domain {
     domain: null,
     _events:
      { removeListener: [Function: updateExceptionCapture],
        newListener: [Function: updateExceptionCapture],
        error: [Function: debugDomainError] },
     _eventsCount: 3,
     _maxListeners: undefined,
     members: [] } }
> { stdout: '6.4.0',
  stderr: '',
  code: 0,
  failed: false,
  killed: false,
  signal: null,
  cmd: 'npm --version',
  timedOut: false }
```

There seems to be something wrong in your test. What was the command?

File: bin/cheminfo-generate.js [hunk 1]
```diff
@@ -3,39 +3,38 @@
...
+      console.error('Not a correct git URL: ' + program.url);
+      return;
     }
-
-    // Yeoman generators
-    const env = yeoman.createEnv();
-    env.register(require.resolve('generator-cheminfo'), 'cheminfo:app');
-    env.run('cheminfo:app', console.error);
-
+    yield execa.shell(`git clone ${program.url}`);
```
Comment (targos) [inline bin/cheminfo-generate.js]: `yield execa('git', ['clone', program.url]);`

File: bin/cheminfo-publish.js [hunk 2]
```diff
@@ -23,245 +22,272 @@ const util = require('../src/util');
...
+  }
+  yield execa('git', ['add', ...filesToAdd]);
+  yield execa('git', ['commit', '-m', newVersion]);
+  yield execa('git', ['tag', '-a', `v${newVersion}`, '-m', `v${newVersion}`]);
+
+  // Publish package
+  console.log('Publishing package');
+  var publishOutput;
+  try {
+    throw new Error('stop here');
```
Comment (targos) [inline bin/cheminfo-publish.js]: wut?

File: bin/cheminfo-publish.js [hunk 3]
```diff
@@ -23,245 +22,272 @@ const util = require('../src/util');
...
+  if (hasChanges) {
+    errorLog(`You have uncommitted changes.`);
+    return;
+  }
+
+  yield git.exec('git pull --rebase');
+
+  // Get npm username
+
+  const name = yield execNpmStdout('whoami');
```
Comment (targos) [inline bin/cheminfo-publish.js:53]: 'execNpm'

File: bin/cheminfo-publish.js [hunk 4]
```diff
@@ -23,245 +22,272 @@ const util = require('../src/util');
...
+      choices: ['mljs', 'cheminfo']
+    })).org;
+  }
+
+  // Get admin list for org
+  if (org === 'cheminfo-js') {
+    org = 'cheminfo';
+  }
+  const adminList = JSON.parse(
+    yield execNpmStdout('team', 'ls', `${org}:developers`)
```
Comment (targos) [inline bin/cheminfo-publish.js]: `execNpm`

File: bin/cheminfo-publish.js [hunk 5]
```diff
@@ -23,245 +22,272 @@ const util = require('../src/util');
...
+      name: 'c',
+      message: 'Are you sure',
+      default: false
+    })).c;
+    if (!confirm) return;
+  }
+
+  // Execute the tests
+  console.log('Running the tests');
+  yield execNpmStdout('run', 'test');
```
Comment (targos) [inline bin/cheminfo-publish.js]: `execNpm`

File: src/generateDoc.js [hunk 6]
```diff
@@ -1,31 +1,43 @@
...
+      '--github',
+      '--output docs',
+      '--format',
+      'html',
+      '--sort-order',
+      'alpha'
+    ]);
+    if (publish) {
+      yield execa('git', ['add', 'docs']);
+      yield execa('git', ['commit', '-m', '"doc: rebuild docs [ci skip]"']);
```
Comment (targos) [inline src/generateDoc.js]: I think you have to remove the double quotes here

**Review by targos — CHANGES_REQUESTED** [review]

File: src/generateDoc.js (hunk 6 above)
Comment (stropitek) [reply src/generateDoc.js]: right, that's the point of having the args as an array

File: bin/cheminfo-publish.js (hunk 2 above)
Comment (stropitek) [reply bin/cheminfo-publish.js]: oups, I was testing it without publishing

File: bin/cheminfo-publish.js (hunk 3 above)
Comment (targos) [reply bin/cheminfo-publish.js:53]: just fix this last one, then you can merge

**Review by targos — APPROVED** [review]

**Pull request merged**